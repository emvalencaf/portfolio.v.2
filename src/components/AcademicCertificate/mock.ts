import { AcademicCertificateProps } from ".";

export default {
	courseTitle: "Curso de React.Js e Next.Js (Intermediário e Avançado)",
	institution: "Udemy, Instrutor: Luiz Otávio Miranda",
	courseWork: "93 horas",
	startIn: 2932983,
	endIn: 3932983,
	urlDownload: "/assets/files/UC-44225967-b7ec-4add-8983-426364ad7bb6.pdf"
} as AcademicCertificateProps;