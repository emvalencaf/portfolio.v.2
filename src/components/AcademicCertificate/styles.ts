import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;