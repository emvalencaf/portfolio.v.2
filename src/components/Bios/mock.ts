import { BiosProps } from ".";

export default {
	bios: `
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore rem minus dolore ducimus quos, sed dolores odit maiores quam tempora atque deserunt voluptas perferendis, aspernatur, sequi dignissimos reiciendis. In, esse!
	Lorem ipsum, dolor sit amet consectetur adipisicing elit. Non a deleniti quia. Magnam aspernatur cumque, nemo vero libero excepturi consequuntur corporis ullam nobis molestiae? Atque nam praesentium ullam quaerat quisquam?
</p>
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio cumque recusandae libero sequi, tempora iste, illo, possimus unde cum numquam quae corrupti minima tenetur mollitia earum repellendus animi molestiae voluptates.
	Lorem ipsum dolor sit, amet consectetur adipisicing elit. Obcaecati dolores fugiat ipsam ut odit sed aliquid ducimus molestiae vero ipsa nam veniam dolor libero labore qui pariatur, alias assumenda itaque?
	Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illo cum temporibus nesciunt sequi odit ipsam harum magnam dolore a molestiae. Totam debitis, rerum aliquid sit non ratione. Et, omnis iusto!
</p>
<p>
	Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nam dicta aut cum sequi voluptatibus. Vel pariatur veritatis quidem. Culpa vero deleniti minima earum quod, odit deserunt fugit reprehenderit architecto in!
</p>
	`,
	profilePhoto: {
		altText: "profile picture",
		srcImg: "assets/images/potrait-photo.jpg",
	},
} as BiosProps;
