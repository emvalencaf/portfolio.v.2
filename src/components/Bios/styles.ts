import styled, { css } from "styled-components";
export const Wrapper = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			flex-direction: column;
		}

		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;
		gap: ${theme.spacings.xhero};
		text-align: justify;
		& p {
			text-align: justify;
			padding: ${theme.spacings.small};
			max-width: 550px;
		}
	`}
`;

export const ContentContainer = styled.div`
	${() => css`
		max-height: 350px;
		overflow-y: auto;
	`}
`;
