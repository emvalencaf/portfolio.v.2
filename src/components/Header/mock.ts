import { HeaderProps } from ".";

export default {
	logo: {
		srcImg: "assets/images/logo.png",
		altText: "logo",
		link: "#",
	},
	menuLinks: [
		{
			children: "home",
			link: "/",
			newTab: false,
		},
		{
			children: "sobre",
			link: "/sobre",
			newTab: false,
		},
		{
			children: "habilidades",
			link: "/habilidades",
			newTab: false,
		},
		{
			children: "projetos",
			link: "/projetos",
			newTab: false,
		},
		{
			children: "curriculo",
			link: "/curriculo",
			newTab: false,
		},
	],
} as HeaderProps;
