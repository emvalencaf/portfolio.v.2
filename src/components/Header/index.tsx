// components
import LogoLink, { LogoLinkProps } from "../LogoLink";
import Menu, { MenuProps } from "../Menu";

// styles
import * as Styled from "./styles";

// types
export type HeaderProps = MenuProps & {
	logo: LogoLinkProps;
	visible?: boolean;
	colorIcon?:
		| "primary"
		| "secondary"
		| "tertiary"
		| "quaternary"
		| "quinary"
		| "senary";
};

const Header = ({
	menuLinks = [],
	logo,
	visible = true,
	colorIcon = "secondary",
}: HeaderProps) => {
	return (
		<Styled.Wrapper visible={visible}>
			<LogoLink {...logo} />
			<Menu menuLinks={menuLinks} colorIcon={colorIcon} />
		</Styled.Wrapper>
	);
};

export default Header;
