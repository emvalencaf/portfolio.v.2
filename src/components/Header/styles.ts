import styled, { css } from 'styled-components';

// imported styled-components
import { Link as LogoLinkContainer } from "../LogoLink/styles";

// types
type ContainerProps = {
	visible: boolean;
}

// function
const visibileMenu = () => css`
	opacity: 0;
	visibility: hidden;
	display: none;
	appearance: none;
`;

export const Wrapper = styled.header<ContainerProps>`
	${({ theme, visible }) => css`
		@media ${theme.media.lteMedium} {
			display: block;
			background-color: transparent;
			border: none;
			& ${LogoLinkContainer} {
				display: none;
				appearance: none;
				opacity: 0;
			}
		}
		position: fixed;
		left: 0;
		top: 0;
		right: 0;
		display: flex;
		align-self: center;
		justify-content: space-between;
		padding-left: ${theme.spacings.medium};
		padding-right: ${theme.spacings.medium};
		background-color: ${theme.colors.gray5};
		border: 1px solid ${theme.colors.gray5};
		border-bottom-left-radius: ${theme.fonts.sizes.small};
		border-bottom-right-radius: ${theme.fonts.sizes.small};
		opacity: 1;
		& ul {
			padding: 0;
		}

		${!visible && visibileMenu()}
	`}
`;
