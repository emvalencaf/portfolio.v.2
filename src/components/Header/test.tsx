import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";

// react component to be test
import Header from ".";

// mock
import mock from "./mock";

describe("<Header />", () => {
	it("should render Header component with menuLink default", () => {
		renderTheme(<Header logo={mock.logo} />);

		expect(screen.getByLabelText("logo link")).toBeInTheDocument();
		expect(
			screen.queryByRole("link", { name: mock.menuLinks[3].link })
		).not.toBeInTheDocument();
	});

	it("should not render Header component", () => {
		renderTheme(<Header {...mock} visible={false} />);

		const header = screen.getByLabelText("logo link").parentElement;

		expect(header).toHaveStyle({
			opacity: 0,
			visibility: "hidden",
			display: "none",
			appearance: "none",
		});
	});

	it("should match snapshot", () => {
		const { container } = renderTheme(<Header {...mock} visible />);

		expect(container).toMatchSnapshot();
	});
});
