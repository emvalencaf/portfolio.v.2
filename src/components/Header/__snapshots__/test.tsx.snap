// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c2 > img {
  width: 6.4rem;
  height: 6.4rem;
  cursor: pointer;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #0A1128;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 350ms ease-in;
  transition: all 350ms ease-in;
}

.c6:hover {
  color: #dc143c;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c5 > ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.6rem;
}

.c5 ul > li {
  list-style: none;
  -webkit-animation: navbarLinkLoad;
  animation: navbarLinkLoad;
  -webkit-animation-duration: 1000ms;
  animation-duration: 1000ms;
  -webkit-animation-delay: 450ms;
  animation-delay: 450ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c5 ul > li:first-child {
  -webkit-animation-name: navbarLinkLoad_first;
  animation-name: navbarLinkLoad_first;
  -webkit-transform: translateX(-1000%);
  -ms-transform: translateX(-1000%);
  transform: translateX(-1000%);
}

.c5 ul > li:last-child {
  -webkit-animation-name: navbarLinkLoad_last;
  animation-name: navbarLinkLoad_last;
  -webkit-transform: translateX(1000%);
  -ms-transform: translateX(1000%);
  transform: translateX(1000%);
}

.c3 {
  display: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: fixed;
  top: 2rem;
  right: 2rem;
  background-color: #0A1128;
  color: #FFFFFF;
  border: none;
  z-index: 6;
  pointer-events: all;
}

.c3 > svg {
  width: 2.4rem;
  height: 2.4rem;
}

.c0 {
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 2.4rem;
  padding-right: 2.4rem;
  background-color: #DDDDDD;
  border: 1px solid #DDDDDD;
  border-bottom-left-radius: 1.6rem;
  border-bottom-right-radius: 1.6rem;
  opacity: 1;
}

.c0 ul {
  padding: 0;
}

@media (max-width:768px) {
  .c5 {
    position: fixed;
    height: 100%;
    width: 100%;
    visibility: hidden;
    opacity: 0;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #FFFFFF;
    -webkit-transition: all 350ms ease-in-out;
    transition: all 350ms ease-in-out;
  }

  .c5 > ul {
    height: 100vh;
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    border: 1px solid #0A1128;
    border-radius: 0.4rem;
    -webkit-appearance: auto;
    -moz-appearance: auto;
    appearance: auto;
    opacity: 1;
  }
}

@media (max-width:768px) {
  .c0 {
    display: block;
    background-color: transparent;
    border: none;
  }

  .c0 .c1 {
    display: none;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    opacity: 0;
  }
}

<div>
  <header
    class="c0"
  >
    <a
      aria-label="logo link"
      class="c1 c2"
      rel="internal"
      target="_self"
    >
      <img
        alt="logo"
        src="assets/images/logo.png"
      />
    </a>
    <button
      aria-label="Open/Close menu"
      class="c3"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <nav
      class="c5"
    >
      <ul>
        <li>
          <a
            class="c6"
            href="/"
            rel="internal"
            target="_self"
          >
            home
          </a>
        </li>
        <li>
          <a
            class="c6"
            href="/sobre"
            rel="internal"
            target="_self"
          >
            sobre
          </a>
        </li>
        <li>
          <a
            class="c6"
            href="/habilidades"
            rel="internal"
            target="_self"
          >
            habilidades
          </a>
        </li>
        <li>
          <a
            class="c6"
            href="/projetos"
            rel="internal"
            target="_self"
          >
            projetos
          </a>
        </li>
        <li>
          <a
            class="c6"
            href="/curriculo"
            rel="internal"
            target="_self"
          >
            curriculo
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
